src/OpenFeature.Hosting/OpenFeatureBuilderExtensions.cs

```diff
@@ -0,0 +1,36 @@
+using Microsoft.Extensions.DependencyInjection;
+
+namespace OpenFeature;
+
+/// <summary>
+/// Extension methods for configuring the hosted feature lifecycle in the <see cref="OpenFeatureBuilder"/>.
+/// </summary>
+public static partial class OpenFeatureBuilderExtensions
+{
+    /// <summary>
+    /// Adds the <see cref="HostedFeatureLifecycleService"/> to the OpenFeatureBuilder, 
+    /// which manages the lifecycle of features within the application. It also allows 
+    /// configuration of the <see cref="FeatureLifecycleStateOptions"/>.
+    /// </summary>
+    /// <param name="builder">The <see cref="OpenFeatureBuilder"/> instance.</param>
+    /// <param name="configureOptions">An optional action to configure <see cref="FeatureLifecycleStateOptions"/>.</param>
+    /// <returns>The <see cref="OpenFeatureBuilder"/> instance.</returns>
+    public static OpenFeatureBuilder AddHostedFeatureLifecycle(this OpenFeatureBuilder builder, Action<FeatureLifecycleStateOptions>? configureOptions = null)
+    {
+        if (configureOptions == null)
+        {
+            builder.Services.Configure<FeatureLifecycleStateOptions>(cfg =>
+            {
+                cfg.StartState = FeatureStartState.Starting;
```

Comment:
Those defaults are already set in property initializers. Additionally this code will override any values which user could set before executing `AddHostedFeatureLifecycle()` (for example bound from `appsettings.json`) because `IConfigureOptions<T>` are executed in order of their registration.

Reply:
Thank you for pointing that out! I’ll review the property initializers and check how this affects the configuration flow, especially with values set from appsettings.json. I appreciate the heads-up and will make adjustments accordingly and get back to you soon.

Reply:
@wwalendz-relativity Are you suggesting replacing the current implementation with something like this:
`builder.Services.AddOptions<FeatureLifecycleStateOptions>();`?
I’m a bit confused, as I believe user-defined configurations, such as those set via appsettings.json, should still take precedence. Could you provide a code snapshot of how you'd suggest replacing the existing code, along with a real-world scenario? This will help clarify how we avoid overriding any user-set configurations unintentionally.